sycl/plugins/unified_runtime/ur/adapters/cuda/context.hpp

```diff
@@ -69,16 +84,23 @@ struct ur_context_handle_t_ {
 
   using native_type = CUcontext;
 
-  native_type CUContext;
-  ur_device_handle_t DeviceID;
+  std::vector<ur_device_handle_t> Devices;
+  uint32_t NumDevices{};
+
   std::atomic_uint32_t RefCount;
 
-  ur_context_handle_t_(ur_device_handle_t_ *DevID)
-      : CUContext{DevID->getContext()}, DeviceID{DevID}, RefCount{1} {
-    urDeviceRetain(DeviceID);
+  ur_context_handle_t_(const ur_device_handle_t *Devs, uint32_t NumDevices)
```

Comment:
It is not possible that the number of devices associated with a context changes across its lifetime, right?

Reply:
Correct. A context is created once either at `q.submit` or at `sycl::context` explicit creation and it can't change afterwards